### ISSUE TITLE
tuple update: introduce append opetaion

### ISSUE THREAD

**Issue by olegrok** [tarantool/tarantool#5152]
Sometimes it could be useful to add element to the end of tuple/nested array and do it atomically. Currently it's impossible because you should know the length of needed tuple/nested array.

I suggest to extend to introduce "append" operation to tuple.update implementation.

**Comment by Gerold103** [comment]
Sounds like better to allow negative indexes in JSON. That would make the existing operations work fine.

**Comment by olegrok** [comment]
> Sounds like better to allow negative indexes in JSON. That would make the existing operations work fine.

Our doc says:
```
For ! and = operations the field number can be -1, meaning the last field in the tuple.
```

If we allow to use negative indexes it could lead to confusion.

**Comment by Gerold103** [comment]
-1 is negative. Also you can use -2, -3 and so on. Out of JSON path. They are allowed already.
If your question was about how ! works with -1 - it works as append: https://github.com/tarantool/tarantool/blob/master/test/box/update.result#L266

**Comment by olegrok** [comment]
> If your question was about how ! works with -1 - it works as append

Yes. I didn't know about that. Then ok, I agree, it would be great to allow negative indexes in JSON-path updates.
